model/testing/src/icon4py/model/testing/data_handling.py

```diff
@@ -10,38 +10,25 @@
 from pathlib import Path
 
 
-def download_and_extract(
-    uri: str, base_path: Path, destination_path: Path, data_file: str = "downloaded.tar.gz"
-) -> None:
+def download_and_extract(uri: str, dst: Path, data_file: str = "downloaded.tar.gz") -> None:
     """
-    "Download data archive from remote server.
+    Download data archive from remote server.
 
-    Checks whether a given directory `destination_path` is empty and if so downloads a the tar
-    file at `uri` and extracts it.
+    Downloads a tar file at `uri` and extracts it.
     Args:
         uri: download url for archived data
-        base_path: the archive is extracted at this path it might be different from the final
-            destination to account for directories in the archive
-        destination_path: final expected location of the extracted data
-        data_file: local final of the archive is removed after download
-
+        destination: the archive is extracted at this path
+        data_file: filename of the downloaded archive, the archive is removed after download
     """
-    destination_path.mkdir(parents=True, exist_ok=True)
```

Comment:
I moved the responsibility of deciding to download or not to the caller of this function.